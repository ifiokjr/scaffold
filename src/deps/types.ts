export * from "https://cdn.skypack.dev/type-fest@2.13.1?dts";
