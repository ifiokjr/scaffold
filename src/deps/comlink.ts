export * from "https://esm.sh/comlink@4?target=deno";
