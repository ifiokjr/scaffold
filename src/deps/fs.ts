export * from "https://deno.land/std@0.144.0/fs/mod.ts";
export * from "https://esm.sh/merge-anything?target=deno";
