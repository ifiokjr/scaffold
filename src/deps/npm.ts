export { render } from "https://deno.land/x/eta@v1.12.3/mod.ts";

export * from "https://esm.sh/is-what@4.1.7?target=deno";
export * from "https://esm.sh/merge-anything@5.0.2?target=deno";
export { default as invariant } from "https://esm.sh/tiny-invariant@1.2.0?target=deno";
export * from "https://esm.sh/ts-extras@0.10.2?target=deno";
