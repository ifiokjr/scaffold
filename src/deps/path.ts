export { default as normalizePath } from "https://cdn.skypack.dev/normalize-path@v3.0.0?dts";
export * as path from "https://deno.land/std@0.144.0/path/mod.ts";
export * from "https://esm.sh/junk@4.0.0?type=deno";
