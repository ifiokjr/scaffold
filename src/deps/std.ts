export { Tar, Untar } from "https://deno.land/std@0.145.0/archive/tar.ts";
export { parse } from "https://deno.land/std@0.145.0/flags/mod.ts";
export { readLines } from "https://deno.land/std@0.145.0/io/mod.ts";
export {
  BaseHandler,
  type HandlerOptions,
} from "https://deno.land/std@0.145.0/log/handlers.ts";
export {
  type LevelName,
  LogLevels,
} from "https://deno.land/std@0.145.0/log/levels.ts";
export { Logger, LogRecord } from "https://deno.land/std@0.145.0/log/logger.ts";
export {
  globToRegExp,
  isGlob,
} from "https://deno.land/std@0.145.0/path/glob.ts";
export {
  copy,
  iterateReader,
  readableStreamFromIterable,
  readableStreamFromReader,
  readerFromIterable,
  readerFromStreamReader,
} from "https://deno.land/std@0.145.0/streams/conversion.ts";
export {
  assert,
  equal,
} from "https://deno.land/std@0.145.0/testing/asserts.ts";
