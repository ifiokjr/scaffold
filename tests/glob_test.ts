import { glob } from "../mod.ts";
import { path } from "../src/deps/path.ts";
import { assertSnapshot, describe, it } from "./deps.ts";

const cwd = path.fromFileUrl(new URL("__fixtures__/glob/", import.meta.url));
function snapshot(t: Deno.TestContext, array: string[]) {
  return assertSnapshot(t, array, {
    serializer: (actual) =>
      Deno.inspect(actual.sort(), {
        colors: false,
        depth: 100,
        iterableLimit: Infinity,
        strAbbreviateSize: 100_000,
        trailingComma: true,
      }),
  });
}

describe("glob", () => {
  it("should return all entries by default", async (t) => {
    const gathered: string[] = [];

    for await (const entry of glob({ cwd })) {
      gathered.push(entry.relative);
    }

    await snapshot(t, gathered);
  });

  it("should accept a function as a matcher", async (t) => {
    const gathered: string[] = [];

    for await (
      const entry of glob({
        include: (p) => !p.startsWith("first"),
        cwd,
        includeDirectories: false,
      })
    ) {
      gathered.push(entry.relative);
    }

    await snapshot(t, gathered);
  });

  it("should return no entries with empty array", async (t) => {
    const gathered: string[] = [];

    for await (const entry of glob({ include: [], cwd })) {
      gathered.push(entry.relative);
    }

    await snapshot(t, gathered);
  });

  it("returns the filtered entries when `include` option provided", async (t) => {
    const gathered: string[] = [];
    const include = ["**/*/*.md"];

    for await (
      const entry of glob({ include, cwd, includeDirectories: false })
    ) {
      gathered.push(entry.relative);
    }

    await snapshot(t, gathered);
  });

  it("should support returning `onlyDirectories`", async (t) => {
    const gathered: string[] = [];

    for await (const entry of glob({ includeFiles: false, cwd })) {
      gathered.push(entry.relative);
    }

    await snapshot(t, gathered);
  });
});
