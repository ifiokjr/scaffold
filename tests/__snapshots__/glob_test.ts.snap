export const snapshot = {};

snapshot[`glob > should return all entries by default 1`] = `
[
  "./",
  "file.md",
  "first/",
  "first/file.md",
  "first/nested/",
  "first/nested/directory/",
  "first/nested/directory/file.json",
  "first/nested/directory/file.md",
  "first/nested/file.md",
  "second/",
  "second/file.md",
  "second/nested/",
  "second/nested/directory/",
  "second/nested/directory/file.md",
  "second/nested/file.md",
]
`;

snapshot[`glob > should accept a function as a matcher 1`] = `
[
  "file.md",
  "second/file.md",
  "second/nested/directory/file.md",
  "second/nested/file.md",
]
`;

snapshot[`glob > should return no entries with empty array 1`] = `[]`;

snapshot[`glob > returns the filtered entries when \`include\` option provided 1`] = `
[
  "first/file.md",
  "first/nested/directory/file.md",
  "first/nested/file.md",
  "second/file.md",
  "second/nested/directory/file.md",
  "second/nested/file.md",
]
`;

snapshot[`glob > should support returning \`onlyDirectories\` 1`] = `
[
  "./",
  "first/",
  "first/nested/",
  "first/nested/directory/",
  "second/",
  "second/nested/",
  "second/nested/directory/",
]
`;
